### PR TITLE
[Backend] Add REST API endpoints for tests

### PR DESCRIPTION
**Pull request by aobolensk**

### CODE REVIEW

File: webforces/api/tests.py [hunk 1]
```diff
@@ -153,13 +153,45 @@ def testGetAlgByTitleEndpoint(self):
...
+        result = response.json()
+        self.assertEqual(result["alg_id"], 1)
+        self.assertEqual(result["title"], "algorithm1")
+        self.assertEqual(result["author_id"], 1)
+        self.assertEqual(result["source"], 'print("Hello")')
+        self.assertEqual(result["tests_id"], [])
+        self.assertEqual(result["lang_id"], 0)
+
+    def testGetTestByAlgIdEndpoint(self):
+        user = User(0, "LOGIN_USER1", "FN_USER1", "SN_USER1", "MN_USER1", [])
```
Comment (skonnov) [inline webforces/api/tests.py]: Add initialization for new fields

File: webforces/api/tests.py [hunk 2]
```diff
@@ -153,13 +153,45 @@ def testGetAlgByTitleEndpoint(self):
...
+        self.core.db.addUser(user)
+        alg = Algorithm(0, "algorithm1", 1, 'print("Hello")', [], 0)
```
Comment (skonnov) [inline webforces/api/tests.py]: Same here

File: webforces/api/tests.py [hunk 3]
```diff
@@ -153,13 +153,45 @@ def testGetAlgByTitleEndpoint(self):
...
+        self.assertEqual(response.status_code, 200)
+        result = response.json()
+        self.assertTrue(str(test.test_id) in list(result.keys()))
+        self.assertEqual(result[str(test.test_id)]["test_id"], test.test_id)
+        self.assertEqual(result[str(test.test_id)]["alg_id"], test.alg_id)
+        self.assertEqual(result[str(test.test_id)]["title"], "Test 1")
+        self.assertEqual(result[str(test.test_id)]["source"], "1 2 3")
+
+    def testGetTestByAlgTitleEndpoint(self):
+        user = User(0, "LOGIN_USER1", "FN_USER1", "SN_USER1", "MN_USER1", [])
```
Comment (skonnov) [inline webforces/api/tests.py]: And here

File: webforces/api/tests.py [hunk 4]
```diff
@@ -153,13 +153,45 @@ def testGetAlgByTitleEndpoint(self):
...
+        self.core.db.addUser(user)
+        alg = Algorithm(0, "algorithm1", 1, 'print("Hello")', [], 0)
```
Comment (skonnov) [inline webforces/api/tests.py]: And for this one

File: webforces/api/tests.py [hunk 5]
```diff
@@ -262,3 +294,23 @@ def testGetAlgByTitleEndpoint(self):
         self.core.db.addAlg(alg)
         response = self.client.get('/api/algs/algorithm1/')
         self.assertEqual(response.status_code, 403)
+
+    def testGetTestByAlgIdEndpoint(self):
+        user = User(0, "LOGIN_USER1", "FN_USER1", "SN_USER1", "MN_USER1", [])
```
Comment (skonnov) [inline webforces/api/tests.py]: here...

File: webforces/api/tests.py [hunk 6]
```diff
@@ -262,3 +294,23 @@ def testGetAlgByTitleEndpoint(self):
...
+        self.core.db.addUser(user)
+        alg = Algorithm(0, "algorithm1", 1, 'print("Hello")', [], 0)
```
Comment (skonnov) [inline webforces/api/tests.py]: here?

File: webforces/api/tests.py [hunk 7]
```diff
@@ -262,3 +294,23 @@ def testGetAlgByTitleEndpoint(self):
...
+        self.core.db.addAlg(alg)
+        test = Test(0, alg.alg_id, "Test 1", "1 2 3")
+        self.core.db.addTest(test)
+        response = self.client.get(f'/api/algs/{alg.alg_id}/tests/')
+        self.assertEqual(response.status_code, 403)
+
+    def testGetTestByAlgTitleEndpoint(self):
+        user = User(0, "LOGIN_USER1", "FN_USER1", "SN_USER1", "MN_USER1", [])
```
Comment (skonnov) [inline webforces/api/tests.py]: Here!

File: webforces/api/tests.py [hunk 8]
```diff
@@ -262,3 +294,23 @@ def testGetAlgByTitleEndpoint(self):
...
+        self.core.db.addUser(user)
+        alg = Algorithm(0, "algorithm1", 1, 'print("Hello")', [], 0)
```
Comment (skonnov) [inline webforces/api/tests.py]: And finally here.

**Review by skonnov — CHANGES_REQUESTED** [review]

File: webforces/api/tests.py (hunk 1 above)
Comment (aobolensk) [reply webforces/api/tests.py]: Done

File: webforces/api/tests.py (hunk 2 above)
Comment (aobolensk) [reply webforces/api/tests.py]: Done

File: webforces/api/tests.py (hunk 3 above)
Comment (aobolensk) [reply webforces/api/tests.py]: Done

File: webforces/api/tests.py (hunk 4 above)
Comment (aobolensk) [reply webforces/api/tests.py]: Done

File: webforces/api/tests.py (hunk 5 above)
Comment (aobolensk) [reply webforces/api/tests.py]: Done

File: webforces/api/tests.py (hunk 6 above)
Comment (aobolensk) [reply webforces/api/tests.py]: Done

File: webforces/api/tests.py (hunk 7 above)
Comment (aobolensk) [reply webforces/api/tests.py]: Done

File: webforces/api/tests.py (hunk 8 above)
Comment (aobolensk) [reply webforces/api/tests.py]: Done

**Review by skonnov — APPROVED** [review]

**Review by megarungle — APPROVED** [review]

**Pull request merged**